skimage/transform/_geometric.py

```diff
@@ -7,6 +7,24 @@
                              _to_ndimage_mode)
 
 
+def _affine_matrix_from_vector(v):
+    """Affine matrix from linearized (d, d + 1) matrix entries."""
+    nparam = v.size
+    # solve for d in: d * (d - 1) = nparam
+    d = (1 + np.sqrt(1 + 4 * nparam)) / 2 - 1
+    dimensionality = int(d)
```

Comment:
Well, there's no such thing as 9 parameters for the affine... It's 6 (2D), 12 (3D), 20 (4D), ... And anyway you've made a mistake typing stuff in somewhere because I get 4.0 for n=20 and 3.88748 for 19?

*But*, it's totally true that one should never trust floating point arithmetic to be exact. I'll use np.round here.